### PR TITLE
Adds customizable "done" status markers and do some cleanup.

### PR DESCRIPTION
**Pull request by ErikaRS**
The cleanup is fixing some tests, updating the README, and updating dependencies.

Some details:
- I don't know why the README isn't diffing cleanly. The change should be limited to 
  - adding documentation for customizable statuses
  - adding some additional content that was missing
  - fixing an out of date section referen ce
- Changes to packages.json are dependency upgrades to the versions recommended by npm
- Changes to existing tests are based on inspecting the code and the other tests
  - Since the tests themselves had sometimes mixed expectations, I decided to treat the code as the source of truth

### CODE REVIEW

**Comment by lumoe** [comment]
What do you think about adding tests for emojis, non-printable chars and regex special characters? I feel like people will add a ton of crazy things into this checkbox.

This is an AI generated example. We should also test for nested tasks etc. 
```
test("get todos with comprehensive status marker edge cases", () => {
  // GIVEN
  const lines = [
    "- [ ] Normal task",
    // Emojis and symbols
    "- [✅] Checkmark emoji",
    "- [❌] Cross emoji",
    "- [✔️] Heavy checkmark",
    "- [✓] Checkmark symbol",
    "- [✗] Ballot X",
    "- [👍] Thumbs up",
    // Control and non-printable characters
    "- [\u0000] Null",
    "- [\u0007] Bell",
    "- [\u0008] Backspace",
    "- [\u001B] Escape",
    // Special Unicode
    "- [\u202E] RTL override",
    "- [\u200B] Zero-width space",
    "- [\u200C] Zero-width non-joiner",
    "- [\u200D] Zero-width joiner",
    // Combining characters
    "- [a\u0300] Letter with accent",
    "- [e\u0301] Letter with acute",
    // Regex special characters
    "- [.] Dot",
    "- [*] Star",
    "- [+] Plus",
    "- [?] Question",
    "- [(] Open paren",
    "- [)] Close paren",
    "- [[] Open bracket",
    "- []] Close bracket",
    "- [{] Open brace",
    "- [}] Close brace",
    "- [^] Caret",
    "- [$] Dollar",
    "- [|] Pipe",
    "- [\\] Backslash",
    "- [/] Forward slash",
    // Mixed cases
    "- [✅\u200B\u0300] Mixed emoji and special",
    "- [.*+?()] Mixed regex",
    "- [a\u0300\u200B] Mixed combining",
  ];

  // WHEN - using all types of characters as done markers
  const todos = getTodos({
    lines,
    doneStatusMarkers:
      "✅❌✔️✓✗👍\u0000\u0007\u0008\u001B\u202E\u200B\u200C\u200Da\u0300e\u0301.*+?()[]{}\\^$|/",
  });

  // THEN - only the normal task should be returned
  const result = ["- [ ] Normal task"];
  expect(todos).toStrictEqual(result);
});
```

**Review by lumoe — CHANGES_REQUESTED** [review]

File: package.json [hunk 1]
```diff
@@ -18,12 +18,15 @@
   "author": "Lukas Mölschl",
   "license": "MIT",
   "devDependencies": {
+    "@jridgewell/sourcemap-codec": "^1.5.0",
```
[the remark this answers is not in the document]
Comment (ErikaRS) [reply package.json:21]: Removing all of them.

**Comment by ErikaRS** [comment]
I added the test changes you suggested. They found that multipoint Unicode graphemes were not being handled correctly. I updated isTodo to handle the more complex parsing needed to handle these characters (with the help of an AI coding assistant, Amp Code (disclaimer: my team builds it)).

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (lumoe) [reply package.json]: I spotted it by looking at the diff and that it was never used. I do not have any automated checks in place to spot these things.
